### PR TITLE
JAMES-2090 Replace Cassandra related constants by configuration

### PR DESCRIPTION
**Pull request by chibenwa**
Sorry, it's a lot of wiring...

### CODE REVIEW

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraUtils.java [hunk 1]
```diff
@@ -22,22 +22,35 @@
...
 
+import org.apache.james.backends.cassandra.CassandraConfiguration;
+
 import com.datastax.driver.core.ResultSet;
 import com.datastax.driver.core.Row;
 
 public class CassandraUtils {
 
-    private static final int FETCH_NEXT_PAGE_ADVANCE_IN_ROW = 100;
+    public static final CassandraUtils DEFAULT_CASSANDRA_UTILS = new CassandraUtils(CassandraConfiguration.DEFAULT_CONFIGURATION);
```
Comment (aduprat) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraUtils.java]: s/DEFAULT_CASSANDRA_UTILS/WITH_DEFAULT_CONFIGURATION

File: backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+            .uidMaxRetry(uidMaxRetry)
+            .fetchNextPageInAdvanceRow(fetchNextPageInAdvanceRow)
+            .flagsUpdateMessageMaxRetry(flagsUpdateMessageMaxRetry)
+            .flagsUpdateMessageIdMaxRetry(flagsUpdateMessageIdMaxRetry)
+            .flagsUpdateChunkSize(flagsUpdateChunkSize)
+            .messageReadChunkSize(messageReadChunkSize)
+            .expungeChunkSize(expungeChunkSize)
+            .build();
+
+        assertThat(configuration.getAclMaxRetry()).isEqualTo(aclMaxRetry);
```
Comment (strokyl) [inline backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/CassandraConfigurationTest.java]: You could use soft assertion here.
```
    @Rule
    public final JUnitSoftAssertions softly = new JUnitSoftAssertions();
```

File: server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java [hunk 3]
```diff
@@ -243,6 +253,43 @@ private AsyncRetryExecutor provideAsyncRetryExecutor(ScheduledExecutorService sc
         return new AsyncRetryExecutor(scheduler);
     }
 
+    @VisibleForTesting
+    @Provides
+    @Singleton
+    CassandraConfiguration provideCassandraConfiguration(CassandraSessionConfiguration sessionConfiguration) throws ConfigurationException {
+        PropertiesConfiguration propertiesConfiguration = sessionConfiguration.getConfiguration();
+
+        return CassandraConfiguration.builder()
```
Comment (strokyl) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java]: We already have a similar discussion and I do not remember the conclusion of it but here there is some logic duplication.
This is the builder responsability to choose for the default value not for this provideCassandraConfiguration.

Comment (chibenwa) [inline server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/CassandraSessionModule.java]: OK

File: src/site/xdoc/server/config-cassandra.xml [hunk 4]
```diff
@@ -93,6 +93,24 @@
         <dd>Optional.
           If specified defines the Cassandra <a href="http://docs.datastax.com/en/developer/java-driver/2.1/manual/pooling/">Specify heartbeat timeout</a></dd>
 
+        <dt><strong>mailbox.max.retry.acl</strong></dt>
+        <dd>Optional. Defaults to 1000.<br/> Controls the number of reties upon Cassandra ACL updates.</dd>
```
Comment (strokyl) [inline src/site/xdoc/server/config-cassandra.xml]: Don't you mean `retries`

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java [hunk 5]
```diff
@@ -76,28 +116,27 @@ protected Mappings getUserDomainMappingsInternal(String user, String domain) thr
...
-                    (first, second) -> first.union(second)));
+        Map<String, Mappings> map = executor.execute(retrieveAllMappingsStatement.bind())
+            .thenApply(resultSet -> cassandraUtils.convertToStream(resultSet)
+                .map(row -> new UserMapping(row.getString(USER), row.getString(DOMAIN), row.getString(MAPPING)))
+                .collect(Guavate.toImmutableMap(
+                    UserMapping::asKey,
+                    UserMapping::toMapping,
+                    Mappings::union)))
+            .join();
         return map.isEmpty() ? null : map;
```
Comment (strokyl) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java]: What you @chibenwa ternary expression hatter, let a ternary expression survive in your PR !!!!
Are you ok today ?

Comment (chibenwa) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRecipientRewriteTable.java]: Did not wanted to start a war over again.

File: mpt/app/dependency-reduced-pom.xml [hunk 6]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <artifactId>apache-james-mpt</artifactId>
     <groupId>org.apache.james</groupId>
-    <version>3.0.0-beta6-SNAPSHOT</version>
+    <version>3.1.0-SNAPSHOT</version>
```
Comment (mbaechler) [inline mpt/app/dependency-reduced-pom.xml]: I guess this file should not be commited at all

Comment (chibenwa) [inline mpt/app/dependency-reduced-pom.xml]: For sure.

**Comment by chibenwa** [comment]
test this please

File: backends-common/cassandra/pom.xml [hunk 7]
```diff
@@ -168,6 +168,12 @@
                     <version>${assertj-3.version}</version>
                     <scope>test</scope>
                 </dependency>
+                <dependency>
+                    <groupId>nl.jqno.equalsverifier</groupId>
+                    <artifactId>equalsverifier</artifactId>
+                    <version>1.7.6</version>
+                    <scope>test</scope>
+                </dependency>
```
Comment (strokyl) [inline backends-common/cassandra/pom.xml]: Indent issue!
I don't know why but we have often indent issue with pom.xml I guess it's a mix between space and tab.
When editing pom.xml make sure to reindent the all file

**Comment by chibenwa** [comment]
To that point the PR is green

File: backends-common/cassandra/pom.xml (hunk 7 above)
Comment (chibenwa) [inline backends-common/cassandra/pom.xml]: Uhm do you smell the upcoming merge conflict between our PRs?

**Comment by aduprat** [comment]
test this please

**Pull request merged**